packages/@aws-cdk/aws-cfn-include/package.json

```diff
@@ -0,0 +1,399 @@
+    "@aws-cdk/aws-iot1click": "0.0.0",
+    "@aws-cdk/aws-iotanalytics": "0.0.0",
+    "@aws-cdk/aws-iotevents": "0.0.0",
+    "@aws-cdk/aws-iotthingsgraph": "0.0.0",
+    "@aws-cdk/aws-kinesis": "0.0.0",
+    "@aws-cdk/aws-kinesisanalytics": "0.0.0",
+    "@aws-cdk/aws-kinesisfirehose": "0.0.0",
+    "@aws-cdk/aws-kms": "0.0.0",
+    "@aws-cdk/aws-lakeformation": "0.0.0",
+    "@aws-cdk/aws-lambda": "0.0.0",
+    "@aws-cdk/aws-lambda-destinations": "0.0.0",
+    "@aws-cdk/aws-lambda-event-sources": "0.0.0",
+    "@aws-cdk/aws-lambda-nodejs": "0.0.0",
+    "@aws-cdk/aws-logs": "0.0.0",
+    "@aws-cdk/aws-logs-destinations": "0.0.0",
+    "@aws-cdk/aws-managedblockchain": "0.0.0",
+    "@aws-cdk/aws-mediaconvert": "0.0.0",
+    "@aws-cdk/aws-medialive": "0.0.0",
+    "@aws-cdk/aws-mediastore": "0.0.0",
+    "@aws-cdk/aws-msk": "0.0.0",
+    "@aws-cdk/aws-neptune": "0.0.0",
+    "@aws-cdk/aws-networkmanager": "0.0.0",
+    "@aws-cdk/aws-opsworks": "0.0.0",
+    "@aws-cdk/aws-opsworkscm": "0.0.0",
+    "@aws-cdk/aws-pinpoint": "0.0.0",
+    "@aws-cdk/aws-pinpointemail": "0.0.0",
+    "@aws-cdk/aws-qldb": "0.0.0",
+    "@aws-cdk/aws-ram": "0.0.0",
+    "@aws-cdk/aws-rds": "0.0.0",
+    "@aws-cdk/aws-redshift": "0.0.0",
+    "@aws-cdk/aws-robomaker": "0.0.0",
+    "@aws-cdk/aws-route53": "0.0.0",
+    "@aws-cdk/aws-route53-patterns": "0.0.0",
+    "@aws-cdk/aws-route53-targets": "0.0.0",
+    "@aws-cdk/aws-route53resolver": "0.0.0",
+    "@aws-cdk/aws-s3": "0.0.0",
+    "@aws-cdk/aws-s3-assets": "0.0.0",
+    "@aws-cdk/aws-s3-deployment": "0.0.0",
+    "@aws-cdk/aws-s3-notifications": "0.0.0",
+    "@aws-cdk/aws-sagemaker": "0.0.0",
+    "@aws-cdk/aws-sam": "0.0.0",
+    "@aws-cdk/aws-sdb": "0.0.0",
+    "@aws-cdk/aws-secretsmanager": "0.0.0",
+    "@aws-cdk/aws-securityhub": "0.0.0",
+    "@aws-cdk/aws-servicecatalog": "0.0.0",
+    "@aws-cdk/aws-servicediscovery": "0.0.0",
+    "@aws-cdk/aws-ses": "0.0.0",
+    "@aws-cdk/aws-ses-actions": "0.0.0",
+    "@aws-cdk/aws-sns": "0.0.0",
+    "@aws-cdk/aws-sns-subscriptions": "0.0.0",
+    "@aws-cdk/aws-sqs": "0.0.0",
+    "@aws-cdk/aws-ssm": "0.0.0",
+    "@aws-cdk/aws-stepfunctions": "0.0.0",
+    "@aws-cdk/aws-stepfunctions-tasks": "0.0.0",
+    "@aws-cdk/aws-transfer": "0.0.0",
+    "@aws-cdk/aws-waf": "0.0.0",
+    "@aws-cdk/aws-wafregional": "0.0.0",
+    "@aws-cdk/aws-wafv2": "0.0.0",
+    "@aws-cdk/aws-workspaces": "0.0.0",
+    "@aws-cdk/cdk-assets-schema": "0.0.0",
+    "@aws-cdk/cloud-assembly-schema": "0.0.0",
+    "@aws-cdk/core": "0.0.0",
+    "@aws-cdk/custom-resources": "0.0.0",
+    "@aws-cdk/cx-api": "0.0.0",
+    "@aws-cdk/region-info": "0.0.0"
+  },
+  "devDependencies": {
+    "@aws-cdk/assert": "0.0.0",
+    "@types/fs-extra": "^8.0.1",
+    "@types/jest": "^25.1.2",
+    "@types/yaml": "1.2.0",
+    "cdk-build-tools": "0.0.0",
+    "jest": "^24.9.0",
+    "pkglint": "0.0.0",
+    "ts-jest": "^25.2.0"
+  },
+  "bundledDependencies": [
+    "fs-extra",
+    "yaml"
```

Comment:
Please avoid these dependencies. Given monocdk and security concerns, the bar for bundling additional dependencies is very high now.

Reply:
We can get rid of `fs-extra` by taking on some inconveniences. But there's no way we can dispose of `yaml` - we will want to add support for YAML CloudFormation templates at some point.

BTW, I don't think this module should be included in monocdk - I think it should be a separate package.

Reply:
How many transitive dependencies are we getting from `yaml`? Does it have any?

I am okay with not including this as part of monocdk, but we currently don't have any provisions for that.

@MrArnoldPalmer maybe we should include an exclusion list of modules in the monocdk builder?

Reply:
Looks like this might prove necessary. Seems like it could be tricky but I have no immediate reason coming to mind as to why.

I'll investigate this as part of the next round of build tooling fixes for the RFC.

Reply:
There's already some (simple) support for excluding: https://github.com/aws/aws-cdk/blob/17e1573d8a13d1f365947898dcc0762406faa207/packages/monocdk-experiment/gen.js#L8-L8

I've added `aws-cfn-include` to that list.

Reply:
As for YAML, it has [one dependency](https://www.npmjs.com/package/yaml/v/1.7.2), which in turn has [one dependency](https://www.npmjs.com/package/@babel/runtime), which in turn has [0 dependencies](https://www.npmjs.com/package/regenerator-runtime).

So, 3 dependencies total in the closure for this one declared dependency.

Reply:
> We can get rid of `fs-extra` by taking on some inconveniences. But there's no way we can dispose of `yaml` - we will want to add support for YAML CloudFormation templates at some point.
> 
> 
> 
> BTW, I don't think this module should be included in monocdk - I think it should be a separate package.

I’ve been thinking about it more and I am wondering: why not make this part of monocdk and deprecate our current `CfnInclude`? Any reason we don’t want to offer this as part of the framework?

Reply:
I just don't see a reason to ship this to every single CDK customer, and every single CDK third-party library, even if they're not using it.

It's a matter of adding a single dependency to your `package.json`, so I don't think it's a big deal.